### DISCUSSION TITLE
How to have multiple self-hosted Supabase instances on a single VM

### DISCUSSION THREAD

**Discussion by Tejaswgupta** [supabase/supabase#22649] [Questions]
We've self hosted Supabase on our VM and it's been going pretty good. We have a couple of projects going on and therefore can't dump all the data into the single Postgres database.

Is it possible to:
- Have multiple database and respective Admin UI 
OR
- Host multiple instances of Supabase

**Comment by M7MMAD-OMAR** [comment]
The official docker compose setup assumes one project per stack, so the usual route is one full copy per project: its own directory and `.env`, a unique compose project name (`docker compose -p app1 up -d`), unique `container_name` values and unique host ports. It works, but every copy runs the whole set of containers, so memory runs out after a few projects.

Disclosure: I built Sbarbase, an open source layer for exactly this. It keeps the original Supabase images and changes the shape underneath:

- **One Postgres cluster**, with a separate database and scoped logins per project environment, so data, users and roles never mix.
- **Original Auth and PostgREST per environment**, one shared Storage, one gateway that checks each request's key and sends it to the right environment.
- **Upstream Studio per environment, on demand**, so each project keeps its own admin UI without a Studio running all the time.

Adding a project costs a database and two small services instead of a full stack, and the server tells you when it has no room left instead of overloading. Existing projects move in with one command (Cloud or self-hosted: tables, policies, users with their passwords, Storage files).

https://github.com/M7MMAD-OMAR/sbarbase (40 second overview: https://base.sbarah.com/)
